src/js/script.js

```diff
@@ -0,0 +1,211 @@
+  }
+}
+function miniclock(secs) {
+  var min = Math.floor(secs / 60);
+  var sec = Math.floor(secs % 60);
+  document.getElementById("count_down").innerHTML =
+    padzero(min) + ":" + padzero(sec);
+}
+
+///***************************
+
+function getsessioninfo(){
+  //get and display latest setting on load 
+  var work_insec = localStorage.getItem('work_ls') || 25*60;
+  var sbreak_insec = localStorage.getItem('sbreak_ls') || 5*60;
+  var lbreak_insec = localStorage.getItem('lbreak_ls') || 15*60;
+  var work_periods = localStorage.getItem('periods_ls') || 4;
+  document.getElementById("worktime").innerText = "Work Session: "+ work_insec/60 + "mins";
+  document.getElementById("shortbreak").innerText = "Short Break: "+ sbreak_insec/60 + "mins";
+  document.getElementById("longbreak").innerText = "Long Break: "+ lbreak_insec/60 + "mins";
+  document.getElementById("workp").innerText = "Work Periods: "+ work_periods;
+  document.getElementById("pomo_length").value = work_insec/60;
+  document.getElementById("sbreak_length").value = sbreak_insec/60;
+  document.getElementById("lbreak_length").value = lbreak_insec/60;
+  document.getElementById("work_periods").value = work_periods;
+}
+
+// On page load, get all session info
+var work_insec = localStorage.getItem('work_ls') || 25*60;
+var sbreak_insec = localStorage.getItem('sbreak_ls') || 5*60;
+var lbreak_insec = localStorage.getItem('lbreak_ls') || 15*60;
+var work_periods = localStorage.getItem('periods_ls') || 4;
+
+document.getElementById("currworktime").innerText = "Work Session: "+ work_insec/60 + "mins";
+document.getElementById("currshortbreak").innerText = "Short Break: "+ sbreak_insec/60 + "mins";
+document.getElementById("currlongbreak").innerText = "Long Break: "+ lbreak_insec/60 + "mins";
+document.getElementById("currworkp").innerText = "Work Periods: "+ work_periods;
+
+miniclock(work_insec);
+
+//TODO: total_work = work_incec etc.
+var total_work = work_insec;
+var total_break = sbreak_insec;
+var total_long_break = lbreak_insec;
+var total_periods = work_periods;
+
+total_work = 5;
+total_break = 5;
+total_long_break = 5;
+total_periods = 2;
+
+var intervalID;
+var curr_session = 0;
+// 0 - no session
+// 1 - working
+// 2 - break
+// 4 - pause from work
+// 5 - pause from chill
+
+var curr_workperiod = 0;
+var total_secs = total_work;
+//total_secs is whatever displays on the clock
+
+// desktop notification
+function showNotification (notif_kind) {
+  if(notif_kind==0){
+    new Notification({title: 'Pomodoro Sessions',
+      body: 'Congratulations! All sessions complete.'}).show();
+  }
+  else if(notif_kind==1){
+    new Notification({title: 'Pomodoro Session',
+      body: 'One working session completes.'}).show();
+  }
+  else if(notif_kind==2){
+    new Notification({title: 'Pomodoro Session',
+      body: 'One working session starts.'}).show();
+  }
+  // more notification kinds could be added here
+}
```

Comment:
```suggestion
function showNotification (notif_kind) {
  var msgText;
  if (notif_kind == 0){
    msgText = 'Congratulations! All sessions complete.';
  } else if (notif_kind == 1) {
    msgText = 'Completed one working session.';
  } else if (notif_kind == 2) {
    msgText = 'Starting one working session.';
  } else {
    throw new Error('Unrecognzied notification type.');
  }
  new Notification({
    title: 'Pomodoro 2: Tokyo Drift',
    body: msgText
  }).show();
}
```

Reply:
Rewrites function to limit code reuse, throw error if unknown notification type received, and fix some grammar.